node-manager/src/node.rs

```diff
@@ -275,27 +274,39 @@ impl Node {
             Arc::clone(&channel_manager),
             router,
             Arc::clone(&logger),
-            event_handler,
-            payment::Retry::Attempts(5), // todo potentially rethink
+            event_handler.clone(),
+            payment::Retry::Attempts(5),
         ));
 
+        let background_persister = persister.clone();
+        let background_event_handler = event_handler.clone();
         let background_processor_logger = logger.clone();
-        let background_processor_invoice_payer = invoice_payer.clone();
         let background_processor_peer_manager = peer_man.clone();
         let background_processor_channel_manager = channel_manager.clone();
         let background_chain_monitor = chain_monitor.clone();
-        let gs: GossipSync<_, _, &NetworkGraph, _, Arc<MutinyLogger>> = GossipSync::none();
 
-        let background_processor = BackgroundProcessor::start(
-            persister.clone(),
-            background_processor_invoice_payer.clone(),
-            background_chain_monitor.clone(),
-            background_processor_channel_manager.clone(),
-            gs,
-            background_processor_peer_manager.clone(),
-            background_processor_logger,
-            Some(scorer),
-        );
+        spawn_local(async move {
+            loop {
+                let gs: GossipSync<_, _, &NetworkGraph, _, Arc<MutinyLogger>> = GossipSync::none();
+                let ev = background_event_handler.clone();
+                process_events_async(
+                    background_persister.clone(),
+                    |e| ev.handle_event(e),
+                    background_chain_monitor.clone(),
+                    background_processor_channel_manager.clone(),
+                    gs,
+                    background_processor_peer_manager.clone(),
+                    background_processor_logger.clone(),
+                    Some(scorer.clone()),
+                    |d| async move {
+                        sleep(d.as_millis() as i32).await;
+                        true
```

Comment:
What is this `true` here for? Do we need to handle some instances where it should be false?

Reply:
For the sleeper is has this description
```rust
/// `sleeper` should return a future which completes in the given amount of time and returns a
/// boolean indicating whether the background processing should exit. Once `sleeper` returns a
/// future which outputs true, the loop will exit and this function's future will complete.
```

I am not too sure by when it should exit, we don't have an extra processing that needs to be done so I think just `true` is fine.

Reply:
![image](https://user-images.githubusercontent.com/15256660/208759159-b1f9c9f2-5ba1-47a5-98d5-dad531d24615.jpeg)